pkg/api/pod/util.go

```diff
@@ -675,6 +677,11 @@ func dropDisabledFields(
 		}
 	}
 
+	// If the feature is disabled and not in use, drop Resources at the pod-level from PodSpec.
```

Comment:
nit. Can we move this to a separate function like we have below in 685-687.

Reply:
done

Reply:
I don't see a change here.